test/tests/functional/pbs_eligible_time.py

```diff
@@ -0,0 +1,103 @@
+# Altair.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software - under
+# a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+from tests.functional import *
+
+
+class TestEligibleTime(TestFunctional):
+    """
+    Test suite for eligible time tests
+    """
+
+    def test_qsub_a(self):
+        """
+        Test that jobs requsting qsub -a <time> do not accrue
+        eligible time until <time> is reached
+        """
+        a = {'eligible_time_enable': 'True', 'scheduling': 'False'}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+
+        # Calculate time which qsub -a expects.  The strftime()
+        # will truncate the seconds, so the job will likely start in
+        # 1 min instead of 2.
+        now = int(time.time())
+        now += 120
+        s = time.strftime("%H%M", time.localtime(now))
+
+        J1 = Job(TEST_USER, attrs={ATTR_a: s})
+        jid = self.server.submit(J1)
+        self.server.expect(JOB, {ATTR_state: 'W'}, id=jid)
+
+        self.logger.info("Sleeping until job is out of 'W' state")
+        time.sleep(120)
+        self.server.expect(JOB, {ATTR_state: 'Q'}, id=jid)
+        self.server.expect(JOB, {'eligible_time': 60}, op=LT)
+
+    def test_job_array(self):
+        """
+        Test that a job array switches from accruing eligible time
+        to ineligible time when its last subjob starts running
+        """
+        a = {'eligible_time_enable': 'True', 'log_events': 2047}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+        a = {'resources_available.ncpus': 2}
+        self.server.manager(MGR_CMD_SET, NODE, a, id=self.mom.shortname)
+
+        J1 = Job(TEST_USER, attrs={ATTR_J: '1-3'})
+        J1.set_sleep_time(20)
+        jid = self.server.submit(J1)
+        jid_short = jid.split('[')[0]
+        sjid1 = jid_short + '[1]'
+        sjid2 = jid_short + '[2]'
+        sjid3 = jid_short + '[3]'
+
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=sjid1, extend='t')
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=sjid2, extend='t')
+        self.server.expect(JOB, {ATTR_state: 'Q'}, id=sjid3, extend='t')
+
+        self.logger.info("Sleeping until subjobs end")
+        time.sleep(20)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=sjid3, extend='t')
+        self.server.expect(JOB, {'accrue_type': 1}, id=jid)
+
+        jid_esp = jid.replace('[', '\[').replace(']', '\]')
+        m = jid_esp + ";Accrue type has changed to ineligible_time, "
+        m += "previous accrue type was eligible_time for 2[0123] secs, "
+        m += "total eligible_time=00:00:2[0123]"
+        print m
```

Comment:
Looks like this was for debugging. please remove it or make it better if you still want it.